sdb/commands/zfs/internal/__init__.py

```diff
@@ -39,6 +42,22 @@ def nicenum(num: int, suffix: str = "B") -> str:
     return "{int(num)}Y{suffix}"
 
 
+def gethrtime() -> int:
+    """
+    The equivalent of gethrtime() in Linux SPL.
+
+    Note that there should be a very minor lag in the order of
+    nanoseconds as there is no set of fields that can give us
+    that level of precision at all times.
+    """
+    tkr = sdb.get_object("tk_core").timekeeper.tkr_raw
+    nsec = tkr.xtime_nsec >> tkr.shift
+    tv_sec = sdb.get_object("tk_core").timekeeper.raw_sec + (nsec / NANOSEC)
+    tv_nsec = nsec % NANOSEC
+    hrtime = ((tv_sec * NANOSEC) + tv_nsec)
```

Comment:
It seems odd to convert nsec to secs + nanosecs, only to combine secs and nanosecs.  Not that it really matters, but couldn't this be simplified to something like:
```
sec = sdb.get_object("tk_core").timekeeper.raw_sec
hrtime = sec * NANOSEC + nsec
```

Reply:
good catch! I was reading the linux kernel source trying out different ways of expressing the current time and never came back to refactor this.